tests/integration/components/bs-button-test.js

```diff
@@ -324,4 +324,25 @@ module('Integration | Component | bs-button', function(hooks) {
     await click('button');
     assert.verifySteps(['onClick action'], 'onClick action is fired again after pending click action is settled');
   });
+
+  test('preventConcurrency=false allows onClick action to be fired concurrently', async function(assert) {
+    // assertions are done implicitly by awaiting the expected result
+    assert.expect(0);
+
+    let deferredClickAction = defer();
+    let clickActionExecutionCount = 0;
+
+    this.set('clickAction', () => {
+      clickActionExecutionCount++;
+      return deferredClickAction.promise;
+    });
+    await render(hbs`{{bs-button onClick=clickAction preventConcurrency=false}}`);
+
+    click('button');
+    await waitUntil(() => clickActionExecutionCount === 1);
```

Comment:
Do we really have to `waitUntil` here? Isn't the action handled synchronously, so we can just `assert.equal(clickActionExecutionCount, 1)`?

Reply:
As far as I've learned from my tests the action is not executed in sync. I guess this is related to runloop. Awaiting next runloop has worked for most scenarios but wasn't a stable pattern. I've set up a CodeSandbox to play around with that behavior: https://codesandbox.io/s/my-app-c276f?fontsize=14&initialpath=%2Ftests&module=%2Ftests%2Fintegration%2Fcomponents%2Fmy-component-test.js

Hope the sandbox is working fine. I'm not that used to CodeSandbox but Ember Twiddle feels so outdated.

Reply:
I think the problem is that `await` is missing for the `click()` call!

Forked your sandbox, with that additional `await` the sync test is also working: https://codesandbox.io/s/my-app-3ptr6

Reply:
`await click` is working for the very basic example in CodeSandox but isn't working for testing pending promises. `await click` won't resolve before the application is in settled state:

> Settled generally means that there are no pending timers, no pending waiters, no pending AJAX requests, and no current run loop. However, new settledness metrics may be added and used as they become available.

Or to put it the other way around: `await click()` would await for the promise returned by `onClick` action to resolve. But the test is about testing how the component behaves while this promise is pending. So this won't work out but result in a timeout.

Reply:
> Or to put it the other way around: await click() would await for the promise returned by onClick action to resolve

Are you sure? AFAIK `settled()` does not include arbitrary promises. 

> Settled generally means that there are no pending timers, no pending waiters, no pending AJAX requests, and no current run loop. However, new settledness metrics may be added and used as they become available.

The pending promise is not included in that enumeration, is it?

Reply:
:man_facepalming: Mixed up testing "real world" interim state, which includes network requests and these mocked ones. Replaced `waitUntil` with `await click()`.